ta/pkcs11/src/processing.c

```diff
@@ -406,13 +420,59 @@ enum pkcs11_rc entry_processing_step(struct pkcs11_client *client,
 	if (rc)
 		return rc;
 
+	if (function == PKCS11_FUNCTION_DIGEST) {
+		if (step == PKCS11_FUNC_STEP_UPDATE_KEY) {
+			rc = serialargs_get(&ctrlargs, &key_handle,
+					    sizeof(uint32_t));
+			if (rc)
+				return rc;
+		}
+
+		if (step == PKCS11_FUNC_STEP_UPDATE ||
+		    step == PKCS11_FUNC_STEP_ONESHOT ||
+		    step == PKCS11_FUNC_STEP_FINAL) {
+			rc = serialargs_get(&ctrlargs, &cmd_flags,
+					    sizeof(uint32_t));
+			if (rc)
+				return rc;
+		}
+	}
+
 	if (serialargs_remaining_bytes(&ctrlargs))
 		return PKCS11_CKR_ARGUMENTS_BAD;
 
 	rc = get_active_session(session, function);
 	if (rc)
 		return rc;
 
+	if (function == PKCS11_FUNCTION_DIGEST) {
+		if (step == PKCS11_FUNC_STEP_UPDATE_KEY) {
```

Comment:
same here, maybe `switch` instead of `if`

Reply:
same here:
```c
	if (step == PKCS11_FUNC_STEP_UPDATE_KEY) {
		assert(function == PKCS11_FUNCTION_DIGEST);

		obj = pkcs11_handle2object(key_handle, session);
		if (!obj) {
			rc = PKCS11_CKR_KEY_HANDLE_INVALID;
			goto out;
		}

Reply:
OK